linkerd/core/src/main/scala/io/buoyant/linkerd/ProtocolInitializer.scala

```diff
@@ -86,9 +87,16 @@ abstract class ProtocolInitializer extends ConfigInitializer { initializer =>
         override def close(d: Time) = Future.Unit
       }
 
-      val servable = servers.map { server =>
-        val stackServer = defaultServer.withParams(defaultServer.params ++ server.params)
-        ServerInitializer(protocol, server.addr, stackServer, adapted, server.announce)
+      val servable = servers.map { s =>
+        val stk = s.params[ClearContext.Param] match {
+          case ClearContext.Param(true) => clearServerContext(defaultServer.stack)
```

Comment:
It is a little redundant, but I don't think it's harmful.  I think it's right for the ClearConext module to exhibit the behavior it does of checking whether it's enabled. And I think it's right that we only apply the protocol-specific transformation when ClearContext is enabled.  What may be confusing is using the same Param in both places... but I don't really think it's worth having an additional Param type for the protocol-specific setting.